scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Record.java

```diff
@@ -385,7 +385,7 @@ private Record(ProtocolVersion version, int epoch, long sequenceNumber, long rec
 			}
 			int length = reader.read(LENGTH_BITS);
 
-			if (reader.bitsLeft() < length) {
```

Comment:
This was a bug ? Which made that some "valid" record be discarded ?

Reply:
> This was a bug ? 

Yes.

> Which made that some "valid" record be discarded ?

No, invalid records may have passed the invalid test. If the length was larger than the actual available bytes, but smaller than the available bits (8xbytes). Encrypted records would have been invalidated with that invalid MAC.